### ISSUE TITLE
file not found libstdc++

### ISSUE THREAD

**Issue by erdemyunsel** [RobertSasak/react-native-openalpr#105]
Hello,

When i run the new project with react-native-openalpr i got this error.

ld: file not found: /usr/lib/libstdc++.6.0.9.dylib
clang: error: linker command failed with exit code 1 (use -v to see invocation)

When I did some research on the Internet, I found out that the library is no longer in use.
https://stackoverflow.com/questions/50694822/xcode-10-ios-12-does-not-contain-libstdc6-0-9/50734510

**Comment by kamellababidi** [comment]
Same Here!
+1

**Comment by RobertSasak** [comment]
Quick search in issues mention similar issue with libstdc. https://github.com/RobertSasak/react-native-openalpr/issues/99

Can that be that you are just missing persmissions as stated in PR?
https://github.com/RobertSasak/react-native-openalpr/pull/102

**Comment by EthanSherr** [comment]
Bump on this issue.  I don't think it's related to #99 or #102.  I believe that the issue is, after running pod install in my project (which uses react-native-openalpr) I start to see, `-weak_library /usr/lib/libstdc++.6.0.9.dylib` in OTHER_LDFLAGS in my project's xcworkspace's Pods-MYAPP.debug.xcconfig / release.xcconfig.  Removing these lines lets my build succeed but this is a temporary fix, as doing another `pod install,` will add it again.  I'm not sure where it comes from...  but I am trying to figure things out.

**Comment by RobertSasak** [comment]
So you have a workaround? Who is adding those lines back again after pod install?

**Comment by richiepreece** [comment]
> Bump on this issue. I don't think it's related to #99 or #102. I believe that the issue is, after running pod install in my project (which uses react-native-openalpr) I start to see, `-weak_library /usr/lib/libstdc++.6.0.9.dylib` in OTHER_LDFLAGS in my project's xcworkspace's Pods-MYAPP.debug.xcconfig / release.xcconfig. Removing these lines lets my build succeed but this is a temporary fix, as doing another `pod install,` will add it again. I'm not sure where it comes from... but I am trying to figure things out. Perhaps it's related to older versions of tesseract that are built with libstdc 6.0.9 ?

This workaround works for me. We should keep the issue open until the workaround isn't needed though.

**Comment by arturBrumm** [comment]
What can fix this issue even on pod install is to change the podfile this way:

 ```
post_install do |installer|
    flipper_post_install(installer)
    installer.pods_project.targets.each do |target|
      if target.name == "Pods-[Your Projectname]"
          puts "Updating #{target.name} -weak library usage"
          target.build_configurations.each do |config|
              xcconfig_path = config.base_configuration_reference.real_path
              xcconfig = File.read(xcconfig_path)
              new_xcconfig = xcconfig.sub('-weak_library /usr/lib/libstdc++.6.0.9.dylib', '')
              File.open(xcconfig_path, "w") { |file| file << new_xcconfig }
          end
      end
    end
  end
```

**Comment by richiepreece** [comment]
FWIW, this might be a Big Sur-only issue. When I upgraded to Big Sur I started seeing this issue. I wonder if they've removed the library since it's no longer supported.

**Comment by EthanSherr** [comment]
> What can fix this issue even on pod install is to change the podfile this way:
> 
> ```
> post_install do |installer|
>    flipper_post_install(installer)
>    installer.pods_project.targets.each do |target|
>      if target.name == "Pods-[Your Projectname]"
>          puts "Updating #{target.name} -weak library usage"
>          target.build_configurations.each do |config|
>              xcconfig_path = config.base_configuration_reference.real_path
>              xcconfig = File.read(xcconfig_path)
>              new_xcconfig = xcconfig.sub('-weak_library /usr/lib/libstdc++.6.0.9.dylib', '')
>              File.open(xcconfig_path, "w") { |file| file << new_xcconfig }
>          end
>      end
>    end
> ```

Well that far beats manually editing the xcconfig file.  Thank you for the better-working workaround!  Issue still should remain open, but for now this fixes my app's CI problem

**Comment by EthanSherr** [comment]
> FWIW, this might be a Big Sur-only issue. When I upgraded to Big Sur I started seeing this issue. I wonder if they've removed the library since it's no longer supported.

Yes I think it's an xcode 10 issue, as libstdc++.6.0.9.dylib is now officially gone

**Comment by EthanSherr** [comment]
> So you have a workaround? Who is adding those lines back again after pod install?

Hey there.  So my original observation was that -- after a pod install in my App, I would get this line: `-weak_library /usr/lib/libstdc++.6.0.9.dylib` in my project's target's xcconfig files.  I found that by just removing this substring from the xcconfig of my project, my app would build And still work as intended.  Doing another pod install though of course would re-add this to my xcconfig and ruin my CI and therefore also ruin my day.  I used arturBumm's `cocoapod post install` fix so that I don't break my pod install process!  Works a lot better for me in the short term!
I believe (not 100% sure) that the underlying issue as to why the libstdc++.6.0.9.dylib line is getting added is because one of the static libraries included in ios/RNOpenAlpr.xcodeproj was using libstdc++.6.0.9.dylib.  Not sure but - I think tesseract library is the culprit.  Here's some more info on libstdc++.6.0.9.dylib which helped me arrive at this conclusion https://www.programmersought.com/article/4069276619/
